### PR TITLE
Adds empty list features

### LINKED ISSUE #7: 7. As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works.
## Summary

When a user starts a new list, they need a little guidance on how to get started. Think through what you'd want to see as a user and make some intentional user experience decisions that will guide our users in confidently adding their very first item. 

## Acceptance criteria
- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item
- [x] Hide from view the filter input if the list is empty 
- [x] Allow the sharing of "empty" lists

## Notes
- [The wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-empty.png) is a little short on the "welcoming" part of the story, but otherwise is functionally complete for the purposes of the AC.

### PR DESCRIPTION
**Pull request by ticiadev**
## Description
This code adds a button on the list page which redirects the user to the add item page if the user's list is empty. It also removes from view the filter search if the user's list is empty. Lastly, by adding in a ```null``` item (which shows up invisible), users are able to share "empty" lists and this will prevent accidental deletion of lists in the future.

## Related Issue

Closes #7  

## Acceptance Criteria

- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item
- [x] Hide from view the filter input if the list is empty 
- [x] Allow the sharing of "empty" lists

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |

### Before

image

### After

image

## Testing Steps / QA Criteria

In your terminal, use `git checkout -b tf-yj-empty-list-first-item` to create a new branch and switch over.
Use `git pull origin tf-yj-empty-list-first-item` to bring the changes from remote to local.
`npm start` to start a local instance of our app.
Remove tcl-shopping-list-token from local storage prior to testing.

- On the home page click `Create a new list` which should bring you to the list view. The list view should display `Your shopping list is currently empty.` and a `Add Item` button.
- The filter search bar should be gone on the list view if its an empty list.
- When the `Add Item` button is clicked, it should redirect you to to the add item view.
- Make sure the list is empty and remember the newly generated token then delete it from local storage.
- Refresh the page and input the same token under `Join an existing list` on the home view and click `Join`. You should now be able to join an existing empty list.

### CODE REVIEW

File: src/views/Home.jsx [hunk 1]
```diff
@@ -16,9 +16,18 @@ export function Home({ handleListTokenState }) {
...
-	const handleCreateList = () => {
+	const handleCreateList = async () => {
 		const newToken = generateToken();
-		handleListTokenState(newToken);
+
+		const placeholder = await addItem(newToken, {
+			itemName: null,
+			daysUntilNextPurchase: '0',
+		});
+		if (placeholder) {
```
Comment (mentalcaries) [inline src/views/Home.jsx:26]: I like this implementation! Nicely done!

File: src/views/List.jsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+import { useNavigate } from 'react-router-dom';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const navigate = useNavigate();
+
 	const [searchInput, setSearchInput] = useState('');
 	const [filteredData, setFilteredData] = useState([]);
 
+	const trueData = data.filter((item) => item.name !== null);
```
Comment (mentalcaries) [inline src/views/List.jsx]: This works great to filter out the placeholder!
What are your thoughts on filtering out the placeholder when making the API request?

**Comment by jeremiahfallin** [comment]
This works like I expect it to and the code looks good to me! The only change I'd consider is what @mentalcaries brought up about adjusting the API request.

**Review by jeremiahfallin — APPROVED** [review]

File: src/views/List.jsx (hunk 2 above)
Comment (ticiadev) [reply src/views/List.jsx]: That's interesting and might make things a little cleaner, I'd love to test that out.

**Review by katherineyuneman — APPROVED** [review]
LGTM!  Everything is functioning as it should be.  I like Devin's point as well and I'd be curious to see the placeholder filter happen during the API fetch too.  But either way love the placeholder idea for storing the empty list!

File: src/views/List.jsx (hunk 2 above)
Comment (Yaosaur) [reply src/views/List.jsx]: We found a way to make it work!

**Review by lyleschemmerling — APPROVED** [review]

**Review by 5hraddha — APPROVED** [review]

**Pull request merged**